cudax/include/cuda/experimental/__cuco/detail/__config

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _CUDAX_DETAIL_CONFIG
+#define _CUDAX_DETAIL_CONFIG
+
+#include <cuda/std/version>
+
+#include <nv/target>
+
+#if !defined(__CUDACC_VER_MAJOR__) || !defined(__CUDACC_VER_MINOR__)
+#  error "NVCC version not found"
+#elif __CUDACC_VER_MAJOR__ < 11 || (__CUDACC_VER_MAJOR__ == 11 && __CUDACC_VER_MINOR__ < 5)
+#  error "NVCC version 11.5 or later is required"
+#endif
```

Comment:
We support different cuda compilers, so this should not happen. if there is no way to support other cuda compilers the code must be guarded by `#if _CCCL_CUDA_COMPILER(NVCC)` or `_CCCL_CUDA_COMPILER(NVRTC)`